File: LinkedList/Middle of the Linked List.java
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode curr = head;
        ListNode run = head;

        while(run!=null && run.next!=null){
            curr = curr.next;
            if(run.next!=null) run = run.next.next;
            else run=null;
            
        }
        return curr;
    }
}

File: LinkedList/Middle of the Linked List.cpp
//876. Middle of the Linked List
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int size=0;
        ListNode* temp = head;

        while(temp!=NULL){
            size++;
            temp = temp->next;
        }

        int mid = size/2;
        while(mid){
            head= head->next;
            mid--;
        }

        return head;
    }
};
